samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -213,32 +227,35 @@ public void onJobModelExpired() {
           if (SamzaContainerStatus.NOT_STARTED.equals(status) || SamzaContainerStatus.STARTED.equals(status)) {
             boolean shutdownComplete = false;
             try {
-              LOGGER.info("Shutting down container in onJobModelExpired for processor:" + processorId);
+              LOGGER.info("Shutting down the container: {} of stream processor: {} in onJobModelExpired.", container, processorId);
               container.pause();
               shutdownComplete = jcContainerShutdownLatch.await(taskShutdownMs, TimeUnit.MILLISECONDS);
-              LOGGER.info("ShutdownComplete=" + shutdownComplete);
+              LOGGER.info("Container shutdown completion status is: {} for stream processor: {}.", shutdownComplete, processorId);
+              if (scheduledSamzaContainerFuture != null && !scheduledSamzaContainerFuture.isDone()) {
```

Comment:
There are states where this could be null during execution. Please check if ignoring that is the right option here. If not, let it throw an NPE/IllegalStateException.